### PR TITLE
crimson/os/seastore: avoid empty Transactions by adding explicit flush() call

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by cyx1231st** [comment]
Yeah, the flush interface is better than the implicit workaround https://github.com/ceph/ceph/pull/44256

**Comment by athanatos** [comment]
Added comments relating flush to transaction submission, probably good to review now.

File: src/crimson/os/seastore/transaction_manager.cc [hunk 1]
```diff
@@ -321,6 +321,17 @@ TransactionManager::submit_transaction_direct(
   });
 }
 
+seastar::future<> TransactionManager::flush(OrderingHandle &handle)
+{
+  return handle.enter(write_pipeline.ool_writes
```
Comment (cyx1231st) [inline src/crimson/os/seastore/transaction_manager.cc:326]: Should `flush()` also need to enter `write_pipeline.reserve_projected_usage` ?

Comment (athanatos) [reply src/crimson/os/seastore/transaction_manager.cc]: Yep, good catch.  Repushed.

**Review by cyx1231st — APPROVED** [review]

**Pull request merged**